index.html

```diff
@@ -101,19 +199,623 @@ <h3>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    If EOF packet was received, run the following steps:
+                    <ol>
+                      <li>
+                        Invoke [=ReadableStream/close=] on [=this=].{{TCPSocket/[[readable]]}}.
+                      <li>
+                        Invoke the steps to [=handle closing the readable stream=].
+                    </ol>
+                  <li>
+                    If no errors were encountered run the following steps:
+                    <ol>
+                      <li>
+                        Let |buffer| be a [=new=] {{ArrayBuffer}} created from |bytes|.
+                      <li>
+                        Let |chunk| be a [=new=] {{Uint8Array}} view over |buffer|,
+                        who's length is the length of |bytes|.
+                      <li>
+                        Invoke [=ReadableStream/enqueue=] on
+                        [=this=].{{TCPSocket/[[readable]]}} with |chunk|.
+                    </ol>
+                  <li>
+                    If an operating system error was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{TCPSocket/[[readable]]}} with
+                    a "{{NetworkError}}" {{DOMException}} and invoke the steps
+                    to [=handle closing the readable stream=].
+                  <li>
+                    If |the socket was disconnected|, run the following steps:
+                    <ol>
+                      <li>Reject [=this=].{{TCPSocket/[[closedPromise]]}}.
+                      <li>
+                        Invoke [=ReadableStream/error=] on
+                        [=this=].{{TCPSocket/[[readable]]}} with a "{{NetworkError}}"
+                        {{DOMException}}.
+                      <li>
+                        Invoke the steps to [=handle closing the readable
+                        stream=].
+                    </ol>
+                </ol>
+            </ol>
+          <li>
+            Let |cancelAlgorithm| be the following steps (in parallel):
+            <ol>
+              <li>
+                Invoke the operating system to close the socket.
+              <li>
+                Invoke the steps to [=handle closing the readable stream=].
+            </ol>
+          <li>
+            [=ReadableStream/Set up=] |stream| with
+            <a href="https://streams.spec.whatwg.org/#readablestream-set-up-pullalgorithm">pullAlgorithm</a>
+            set to |pullAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#readablestream-set-up-cancelalgorithm">cancelAlgorithm</a>
+            set to |cancelAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#readablestream-set-up-highwatermark">highWaterMark</a>
+            set to 0.
```

Comment:
The high water mark should be set to the read buffer size specified in the constructor.

Reply:
No longer relevant